docs/events/gsoc/2024/checkstyle-plugin.md

```diff
@@ -1,3 +1,39 @@
 # GSoC 2024. Declarative Syntax and Enhancements for the Checkstyle Plugin
 
-Description is coming soon
+## Goal
+- **Enhanced Functionality**: Enhance the Checkstyle plugin to support more of Checkstyle's features.
+- **Better Declarativity**: improve the declarative nature of Kotlin DSL when integrating plugins.
+
+## Status
+Selected
+
+## Team
+#### Authors
+- [@Hongjie Wei](https://github.com/JayWei1215)
+
+#### Mentors
+- [@oleg-nenashev](https://github.com/oleg-nenashev)
+
+## Background
+- Checkstyle is a powerful tool, but currently, the Checkstyle plugin in Gradle may not support all of Checkstyle's features.
+- Kotlin DSL does not provide good declarative support when it comes to integrating plugins.
+
+## Implementation
+
+- **Enhanced Functionality** - The CheckstyleExtension class is the extension class of the Gradle Checkstyle plugin.It is used to configure various options of the Checkstyle plugin.By Creating a new subclass of CheckstyleExtension and add some new properties, we can make the Checkstyle plugin support more of Checkstyle's features.
+
+- **Better Declarativity** - Create a Kotlin class to serve as the entry point for the DSL. This class can inherit from CheckstyleExtension. Within the DSL class, define properties and methods specific to the DSL, allowing for the configuration of Checkstyle's various options.
+
+## Target Repositories
+
+
+## Deliverables
+
+-  Provide an enhanced version of the Gradle Checkstyle plugin. It will support additional Checkstyle features, such as code complexity, duplication, and dependency checks.
```

Comment:
I think we should also add having a fork of the Checkstyle plugin as a first deliverable so that we can deliver faster on the rest w/o relying on Gradle Build Tool release cycles. Something to add to the implementation too IMHO